tests/chainer_tests/functions_tests/pooling_tests/test_unpooling_2d.py

```diff
@@ -0,0 +1,78 @@
+import unittest
+
+import numpy
+import six
+
+import chainer
+from chainer import cuda
+from chainer import functions
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+
+
+@testing.parameterize(*testing.product({
+    'cover_all': [True, False],
+}))
+class TestUnpooling2D(unittest.TestCase):
+
+    def setUp(self):
+        self.N = 2
+        self.n_channels = 3
+        inh, inw = (2, 1)
+        self.x = numpy.arange(
+            self.N * self.n_channels * inh * inw, dtype=numpy.float32)\
+            .reshape(self.N, self.n_channels, inh, inw)
+        numpy.random.shuffle(self.x)
+        self.x = 2 * self.x / self.x.size - 1
+
+        self.ksize = 2
+        outh, outw = self.outsize = (4, 2)
+        self.gy = numpy.random.uniform(
+            -1, 1, (self.N, self.n_channels, outh, outw)).astype(numpy.float32)
+
+    def check_forward(self, x_data):
+        x = chainer.Variable(x_data)
+        y = functions.unpooling_2d(x, self.ksize, outsize=self.outsize,
```

Comment:
Could you add test cases where `self.outsize` is different from the result of `get_deconv_outsize`?